lms/djangoapps/certificates/signals.py

```diff
@@ -25,21 +29,9 @@
 
 @receiver(post_save, sender=CertificateWhitelist, dispatch_uid="append_certificate_whitelist")
 def _listen_for_certificate_whitelist_append(sender, instance, **kwargs):  # pylint: disable=unused-argument
-    switches = waffle.waffle()
-    # No flags enabled
-    if (
-        not switches.is_enabled(waffle.SELF_PACED_ONLY) and
-        not switches.is_enabled(waffle.INSTRUCTOR_PACED_ONLY)
-    ):
+    if not auto_certificate_generation_enabled_for_course(courses.get_course_by_id(instance.course_id, depth=0)):
```

Comment:
Either in this PR or the next one, please change this code to not get the course from the modulestore.  Instead, get the course-overview object from SQL (a single SQL query).  Otherwise, `get_course_by_id` connects to the MongoDB and does a whole shin-bang to re-create the course object.

Alternatively, if you change the code so pacing doesn't matter, then you may not need to get the course here.

Reply:
I've been meaning to ask you: when do CourseOverviews get updated?  Is it contemporaneous with updates to mongo, or are CourseOverviews more like a cache?

I think pacing is almost always going to matter, at least the way we designed our decision matrix.

Reply:
CourseOverviews get updated on Course Publish - a celery task gets fired as soon as any Studio edit happens.

We can view CourseOverviews as a local LMS cache of the course-metadata.

Typically, in all of our LMS code, we no longer want to be dependent on the Modulestore.  Over the last 1-2 years, we've been slowly migrating away from the Modulestore in the LMS.